### ISSUE TITLE
Is there a way to make use of the SerenityBDD - JIRA integration?

### ISSUE THREAD

**Issue by cmurczek** [serenity-js/serenity-js#189]
Is it possible to use the JIRA integration of SerenityBDD with serenity-js? I tried using the configuration as described here (http://serenity-bdd.info/docs/serenity/#_feature_coverage) but no success so far.

I have assigned _@issue_ tags to scenarios In my `.feature` files, but when generating the report serenity won't include the link to (tried specifiying jira.user/password/project/url settings via serenity.properties, as well as _--jiraProject_ and _--jiraUrl_ cli options), nor add pending tests based on jira issues.

**Comment by JGONGSQ** [comment]
Any progress on this feature? Appreciated.

**Comment by abhinaba1080** [comment]
@jan-molak  Please update if you have any progress on that. Many thanks.  Solution should support ccucumber 1.x version (i.e Serenity-js 1)

**Comment by iranicus** [comment]
Having recently decided to give serenity-js (v1) a go for one of our angular 4 projects, this is currently a feature in the older selenium framework I am moving the project away from. It would be handy if it were an option for serenity-js v1 too since it was handy to have the Jira refs in the reports for the client.

**Comment by jan-molak** [comment]
I need to look into enabling the JIRA integration properly.

In the meantime, remember that Serenity treats Cucumber scenario and feature descriptions as markdown, so you could have JIRA links there:

```gherkin
Feature: JIRA Intergration

  Here's the [JIRA epic](https://jira.my.org/....)

  Scenario: Scenario name

    Here is [the ticket](https://jira.my.org/....)

    Given ...
    When ...
    Then ...

```

**Comment by brabenetz** [comment]
> I have assigned @issue tags to scenarios In my .feature files, but when generating the report serenity won't include the link to

I think I found the root-cause, and have a (very ugly) workaround for it.

First I need the cucumber results also as json format **"json:target/cucumber.json"** in my `protractor.conf.js`:

```js
    cucumberOpts: {
        tags: ['~@Ignore', '@Regression'],
        require: [path.join(__dirname, 'src/step_definitions/**/*-steps.ts')],
        format: ["pretty", "json:target/cucumber.json"],
        keepAlive: false
    }
```
But the cucumber tests generates only an "issues"-node into the json-results:
```json
    "issues": [
        "JIRA-123"
    ],
```

The Java-Version of cucumber generates also a "additionalIssues"-node into the json-results:
```json
    "issues": [
        "JIRA-123"
    ],
    "additionalIssues": [
        "JIRA-123"
    ],
```

The Java [serenity-cli](https://mvnrepository.com/artifact/net.serenity-bdd/serenity-cli) (which is called by the serenity-js) seems only to read the "additionalIssues"-node...

--------------------------------------------
Now comes my ugly workaround:
I just patches the results of the tests before run "serenity:run":

--------------------------------------------
**package.json** scripts/dependencies (simplified):
```json
  "scripts": {
    "e2e": "failsafe e2e:rimraf e2e:test e2e:fix-issues-in-report-results e2e:report",
    "e2e:rimraf": "rimraf target/site",
    "e2e:test": "ng e2e --project=testbed --webdriver-update=false",
    "e2e:fix-issues-in-report-results": "node e2e/fix-issues-in-report-results.js",
    "e2e:report": "serenity run"
  },
  "dependencies": {
    "npm-failsafe": "^0.4.1",
    "replace": "1.0.0"
  }
```

**fix-issues-in-report-results.js**:
```js
var replace = require('replace');
replace({
  regex: "\"issues\"",
  replacement: "\"additionalIssues\"",
  paths: ['target/site'],
  recursive: true,
  include: '*.json',
  silent: false,
  quiet: false,
  multiline: true,
});
```
I use a "serenity.properties" to configure the "serenity.issue.tracker.url", but there are many ways to configure that.

**Comment by brabenetz** [comment]
Update: **"json:target/cucumber.json" is not needed**, the json-files per test will allways be written into `target/site/serenity/*.json`
The config  `format: ["pretty"],` is enough:
```js
    cucumberOpts: {
        tags: ['~@Ignore', '@Regression'],
        require: [path.join(__dirname, 'src/step_definitions/**/*-steps.ts')],
        format: ["pretty"],
        keepAlive: false
    }
```

**Comment by jan-molak** [comment]
Right; so what you're saying, @brabenetz is that we need both `issues` and `additionalIssues` in the `json` reports produced by Serenity/JS?

**Comment by brabenetz** [comment]
Yes, seems so... (The java version writes the ids into "issues" & "additionalIssues")
With the regex replace my Jira links (from the "serenity.issue.tracker.url") are now shown on the reports.

**Comment by jan-molak** [comment]
Hi @cmurczek, @cmelion, @san-ouadghiri, @andreasgebauer, @otoniel-isidoro-sofist, @JGONGSQ, @brabenetz, @iranicus and @abhinaba1080!

This feature required introducing changes in Serenity BDD Core, Serenity BDD CLI and Serenity/JS projects so it all took a while to implement. 

Integration with Jira (and other issue trackers, such as Trello) is available as of `@serenity-js/serenity-bdd-reporter` `v2.3.0`, so please take it for a spin and let me know what you think!

I also took some time to describe configuring the integration in a bit more depth in [Serenity/JS Handbook: Integration - Jira and other issue trackers](https://serenity-js.org/handbook/integration/jira-and-other-issue-trackers.html).

By the way, if you have found this feature useful, please consider becoming a [Github Sponsor of Serenity/JS](https://github.com/sponsors/serenity-js) and buying us a coffee every once in a while :octocat:

Best,
Jan

**Comment by brabenetz** [comment]
FYI,
Thanks, it works fine on my Project (it only took me a while to upgrade from serenity-js 1 ot 2).

But my first touch with the new feature did not work that well.
I tried to adapt the serenity-js demo-app: https://github.com/serenity-js/serenity-js-jasmine-protractor-template
 1. features/super_calculator.feature: Add Annotation "@issue:189"
 2. package.json: "test:report": "serenity-bdd run --issueTrackerUrl 'https://github.com/serenity-js/serenity-js/issues/{0}' --features ./features",

The result was a big fat NPE (see serenity-core ticket)!

But the error "only" occurs if the Issue ID is a number. It works fine if the Issue ID would have a prefix like "@issue:XYZ-189".
(With that it is not a Problem on my current Project).

I opened a Ticket in serenity-core: 
https://github.com/serenity-bdd/serenity-core/issues/2049
